flake.nix

```
@@ -3,15 +3,27 @@
 
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    treefmt-nix = {
+      url = "github:numtide/treefmt-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
   };
 
   outputs = {
     self,
     nixpkgs,
+    treefmt-nix,
     ...
   }: let
-    forAllSystems = nixpkgs.lib.genAttrs ["x86_64-linux" "aarch64-linux"];
+    supportedSystems = ["x86_64-linux" "aarch64-linux"];
+
+    forAllSystems = function:
+      nixpkgs.lib.genAttrs
+      supportedSystems
```

Just to make the function a bit more clear:

```suggestion
      nixpkgs.lib.genAttrs supportedSystems
```